src/utils/enums.rs

```diff
@@ -0,0 +1,39 @@
+use serde::Serialize;
+
+#[derive(clap::ValueEnum, Clone, Debug, Serialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum Survey {
+    Ztf,
+    Lsst,
+}
+
+impl Survey {
+    pub fn as_str(&self) -> &str {
+        match self {
+            Survey::Ztf => "ZTF",
+            Survey::Lsst => "LSST",
+        }
+    }
+}
```

Comment:
It's more idiomatic to impl Display when you want to be able to turn things into strings:

```rust
impl std::fmt::Display for Survey {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        let string = match self {
            Survey::Ztf => "ZTF",
            Survey::Lsst => "LSST",
        };
        write!(f, "{}", string)
    }
}
```

Then you can get rid of all of the `as_str()` calls, e.g., in `bin/scheduler.rs`, `info!("initialized indexes for {}", survey.as_str())` -> `info!("initialized indexes for {}", survey)`. In `scheduler::base`, you would need to change `table.get(survey_name.as_str())` -> `table.get(&survey_name.to_string())`.

Reply:
agreed 100%, let me add it